src/main/java/fi/jubic/quanta/controller/ExternalClientController.java

```diff
@@ -17,29 +21,73 @@
 
     private final ExternalClientDao externalClientDao;
     private final TaskDao taskDao;
+    private final UserDao userDao;
 
     @Inject
     ExternalClientController(
             ExternalClientDomain externalClientDomain,
             ExternalClientDao externalClientDao,
-            TaskDao taskDao
+            TaskDao taskDao,
+            UserDao userDao
     ) {
         this.externalClientDomain = externalClientDomain;
         this.externalClientDao = externalClientDao;
         this.taskDao = taskDao;
+        this.userDao = userDao;
     }
 
-    public ExternalClient generateExternalClient(Long taskId, ExternalClient externalClient) {
+    public List<ExternalClient> search(ExternalClientQuery query) {
+        return externalClientDao.search(query);
+    }
+
+    public ExternalClient generateExternalClient(ExternalClientView externalClientView) {
+        User user = userDao

```

Comment:
The current user should be used as the `createdBy` user.

Reply:
used authenticated User